modules/compiler/spirv-ll/test/glsl/CMakeLists.txt

```diff
@@ -806,10 +806,20 @@ foreach(glsl ${GLSL_FILES})
   if(GLSL_UNSUPPORTED STREQUAL False)
     # Generate .spv from .comp source.
     file(RELATIVE_PATH spvOut ${CMAKE_BINARY_DIR} ${spv})
+
+    # Work out whether we're validating this SPIR-V file.
+    set(SPIRV_VAL_CMD "")
+    set(SPIRV_VAL_DEP "")
+    if(TARGET spirv::spirv-val)
+      set(SPIRV_VAL_DEP spirv::spirv-val)
+      set(SPIRV_VAL_CMD spirv::spirv-val ${spv})
+    endif()
+
     add_custom_command(OUTPUT ${spv}
       COMMAND ${glslangValidator_EXECUTABLE} -s -V -o ${spv} ${comp}
-      DEPENDS ${glslangValidator_EXECUTABLE} ${comp}
-      COMMENT "Building SPIR-V ${spvOut}")
+      COMMAND ${SPIRV_VAL_CMD}
```

Comment:
I don't think `add_custom_command` specifies that if a command is empty (`spirv-val` is not found), nothing runs, so I'm a bit cautious about relying on that. Can we avoid that somehow?

Reply:
Hmm good point, I hadn't considered that. I don't really know what else to do as my cmake isn't very good. Adding an additional optional `add_custom_command` would probably necessitate some dummy output?

Reply:
We may be able to set a variable to multiple words, so that `COMMAND` is part of the variable and if no command is found, no second `COMMAND` is included. Not entirely sure that would work though.

Reply:
That might work, yeah. Not sure! For now I've gone with a second custom command creating a timestamp file, and using that as an additional dependency to the set of "lit test files"

Reply:
Hmm, if we have two `add_command`s anyway, we might as well do
```
if(CONDITION)
    add_custom_command(COMMAND command1)
else()
    add_custom_command(COMMAND command1 COMMAND command2)
end()
```
? It looks equal in CMake complexity to what you have currently, but doesn't generate the unneeded stamp files.

Reply:
Fair point. I'd be worried about repetition but maybe it's then a good idea to try the "`COMMAND` inside a variable" thing you suggested earlier. I'm pretty sure we do that elsewhere (not for `COMMAND` but for other cmake arguments)

Reply:
I've updated with the first option that we discussed privately.